qiita_db/software.py

```diff
@@ -86,32 +102,91 @@ def create(cls, software, name, description, cli_cmd, parameters_table):
             The name of the command
         description : str
             The description of the command
-        cli_cmd : str
-            The CLI used to call this command
-        parameters_table : str
-            The name of the table in which the parameters of the commands are
-            stored
+        parameters : dict
+            The description of the parameters that this command received. The
+            format is: {parameter_name: (paramter_type, default)},
+            where parameter_name, paramter_type and default are strings. If
+            default is None,
 
         Returns
         -------
         qiita_db.software.Command
             The newly created command
+
+        Raises
+        ------
+        QiitaDBError
+            - If the parameters dictionary is malformed
+            - If one of the parameter types is not supported
+            - If the default value of a choice parameter is not listed in
+            the available choices
+        QiitaDBDuplicateError
+            - If the command already exists
+
+        Notes
+        -----
+        If the default value for a parameter is NULL, then the parameter will
+        be required. On the other hand, if it is provided, the parameter will
+        be optional and the default value will be used when the user doesn't
+        overwrite it.
         """
+        # Perform some sanity checks in the parameters dictionary
+        sql_param_values = []
+        for pname, vals in parameters.items():
+            if len(vals) != 2:
+                raise qdb.exceptions.QiitaDBError(
+                    "Malformed parameters dictionary, the format should be "
+                    "{param_name: [paramter_type, default]}. Found: "
+                    "%s for parameter name %s" % (vals, pname))
+
+            ptype, dflt = vals
+            # Check that the type is one of the supported types
+            supported_types = ['string', 'integer', 'float', 'artifact',
```

Comment:
The reason why I make the difference is to be able to generate the interface automatically if we want, so we can limit the user input.